lib/error/http_problem.go

```diff
@@ -0,0 +1,139 @@
+	ProblemDefaultServiceUnavailable            = NewStatusProblem(http.StatusServiceUnavailable)
+	ProblemDefaultGatewayTimeout                = NewStatusProblem(http.StatusGatewayTimeout)
+	ProblemDefaultHTTPVersionNotSupported       = NewStatusProblem(http.StatusHTTPVersionNotSupported)
+	ProblemDefaultVariantAlsoNegotiates         = NewStatusProblem(http.StatusVariantAlsoNegotiates)
+	ProblemDefaultInsufficientStorage           = NewStatusProblem(http.StatusInsufficientStorage)
+	ProblemDefaultLoopDetected                  = NewStatusProblem(http.StatusLoopDetected)
+	ProblemDefaultNotExtended                   = NewStatusProblem(http.StatusNotExtended)
+	ProblemDefaultNetworkAuthenticationRequired = NewStatusProblem(http.StatusNetworkAuthenticationRequired)
+)
+
+//  Default Http Problem with new description
+var (
+	// Example
+	ProblemBadRequestNotEnoughParam = NewDetailedStatusProblem(http.StatusBadRequest, "paramaters are not enough")
+)
+
+// Http Problem with description and Instance
+var (
+	// Example
+	ProblemBadRequestNotEnoughParamWithInstance = NewDetailedStatusProblem(http.StatusBadRequest, "paramaters are not enough").SetInstance("blahblah")
+)
+
+type Problem struct {
+	// "type" (string) - A URI reference [RFC3986] that identifies the
+	// problem type.  This specification encourages that, when
+	// dereferenced, it provide human-readable documentation for the
+	// problem type (e.g., using HTML [W3C.REC-html5-20141028]).  When
+	// this member is not present, its value is assumed to be
+	// "about:blank".
+	Type string `json:"type"`
+
+	//"title" (string) - A short, human-readable summary of the problem
+	//type.  It SHOULD NOT change from occurrence to occurrence of the
+	//problem, except for purposes of localization (e.g., using
+	//proactive content negotiation; see [RFC7231], Section 3.4).
+	Title string `json:"title"`
+
+	//"status" (number) - The HTTP status code ([RFC7231], Section 6)
+	//generated by the origin server for this occurrence of the problem.
+	Status int `json:"status,omitempty"`
+
+	//"detail" (string) - A human-readable explanation specific to this
+	//occurrence of the problem.
+	Detail string `json:"detail,omitempty"`
+
+	//"instance" (string) - A URI reference that identifies the specific
+	//occurrence of the problem.  It may or may not yield further
+	//information if dereferenced.
+	Instance string `json:"instance,omitempty"`
+}
+
+func NewStatusProblem(status int) Problem {
+	return Problem{Type: "about:blank", Status: status, Detail: http.StatusText(status)}
+}
+
+func NewDetailedStatusProblem(status int, detail string) Problem {
+	p := NewStatusProblem(status)
+	p.Detail = detail
+	return p
+}
+
+func (p Problem) SetInstance(instance string) Problem {
+	p.Instance = instance
+	return p
+}
+
+func (p Problem) SetDetail(detail string) Problem {
+	p.Detail = detail
+	return p
+}
+
+func (p Problem) Serialize() ([]byte, error) {
```

Comment:
It just suggestion, please feel free. ;) `http.Error` receives writer and error code. How about design `StatusProblem` to be similar 
https://golang.org/pkg/net/http/#Error . It will be zero learning :)